mlx_benchmark/run_benchmark.py

```diff
@@ -23,107 +22,110 @@
 from operations import *
 
 
-def run_processes(operations, args, iterations=5):
+def run_processes(operations, backends, iterations=5):
     """
-    Runs all operations in serial, on separate processes.
-    Using processes avoids exploding memory within the main process during the bench.
+    Runs all operations, on all backends, for a specified number of iterations.
     """
     all_times = defaultdict(dict)
-    queue = mp.Queue()
 
-    with tqdm(total=len(operations) * iterations) as pbar:
-        for op in operations:
-            op_times = defaultdict(list)
-            op_name = None
+    for i, backend in enumerate(backends):
+        print(f"\nRunning benchmarks on {backend} ({i + 1}/{len(backends)})")
+        if 'mlx' in backend:
+            times = run_mlx_backend(operations, backend, iterations)
+        else:
+            times = run_backend(operations, backend, iterations)
+        for op_name, duration in times.items():
+            all_times[op_name][backend] = duration
 
-            for _ in range(iterations):
-                p = mp.Process(target=run, args=(op, args, queue))
-                p.start()
+    print("\nDetailed benchmark:")
+    print_benchmark(all_times, backends)
+    print("\n Average benchmark:")
+    print_benchmark(all_times, backends, reduce_mean=True)
 
-                times = queue.get()
-                p.join()
+def run_mlx_backend(operations, backend, iterations, ops_per_process=10):
```

Comment:
ops_per_process determines the number of ops to run in each process. Decreasing this number decreases the max memory usage.